### PR TITLE
On Demand TV Logging

### PR DESCRIPTION
**Pull request by ryanmccombe**
**Overall change:**
_Implement logging for On Demand TV_

**Code changes:**

- Simplify `pathWithLogging`
- Apply logging to On Demand TV
- Some cleanup of On Demand TV & Radio
- Removing the morph registry URLs from package-lock

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/routes/onDemandTV/getInitialData/index.js [hunk 1]
```diff
@@ -1,90 +1,94 @@
...
+import nodeLogger from '#lib/logger.node';
 
-const getEpisodeAvailability = ({ availableFrom, availableUntil }) => {
+const logger = nodeLogger(__filename);
+
+const getEpisodeAvailability = ({ availableFrom, availableUntil, url }) => {
   const timeNow = Date.now();
-  if (!availableUntil || timeNow < availableFrom) return false;
+  if (!availableUntil || timeNow < availableFrom) {
+    logger.info(TV_EPISODE_EXPIRED, {
```
Comment (karinathomasbbc) [inline src/app/routes/onDemandTV/getInitialData/index.js]: Ideally we would follow the single responsibility principle here, and not combine get episode availability with logging.

Would it be possible to look at what was done for onDemandRadio https://github.com/bbc/simorgh/blob/latest/src/app/routes/onDemandRadio/getInitialData/index.js, and consider refactoring for max reusability?

File: od-tv-logging-patch.diff [hunk 2]
```diff
@@ -0,0 +1,441 @@
+From 7f2bc4c4624cf336f8c3fd01223a828e2d6447b9 Mon Sep 17 00:00:00 2001
```
Comment (karinathomasbbc) [inline od-tv-logging-patch.diff]: Oops, not sure if this should be included here

File: src/app/routes/onDemandTV/getInitialData/index.js (hunk 1 above)
Comment (ryanmccombe) [reply src/app/routes/onDemandTV/getInitialData/index.js]: Most of this function is combining getting and logging, as well as the function itself (`getInitialData`)

There's an alternative here: https://github.com/bbc/simorgh/compare/od-tv-logging...od-tv-logging-a but I think it's an unnecessary complexity

**Review by rebeccamcginn — APPROVED** [review]

File: src/app/routes/onDemandTV/getInitialData/index.js (hunk 1 above)
Comment (karinathomasbbc) [reply src/app/routes/onDemandTV/getInitialData/index.js]: Keen to get more opinions on this

Comment (rebeccamcginn) [reply src/app/routes/onDemandTV/getInitialData/index.js]: The favoured approach on logging regarding the single responsibility principle is to delegate to a logger, which is what is implemented here. The function isn't handling implementation of logging, it's passing that responsibility over to the available logger. I'd say good to go.

Comment (RichardPK) [reply src/app/routes/onDemandTV/getInitialData/index.js]: It rustles me jimmies (Ryan, 2020) that we have different implementations of the same logic for OD-radio & OD-tv here. `getUrl` as well as calling the logger within `getEpisodeAvailability` are essentially replicating what's being done within OD-radio's `logInitialData` file, right?

Maybe this should all be pushed into a `logEpisodeUnavailable` logging util akin to `pathWithLogging` that would then be used by both radio & TV (removing od-radio's `logInitialData`)?

Comment (ryanmccombe) [reply src/app/routes/onDemandTV/getInitialData/index.js]: Alright - I'll check with Jane if the expired episode for TV and Radio is the same (including this issue: https://github.com/bbc/simorgh/issues/6889)

If so, I'll move this function into the `lib` folder after https://github.com/bbc/simorgh/pull/6970 is merged

**Review by karinathomasbbc — APPROVED** [review]
Looks good 👍

**Review by RichardPK — APPROVED** [review]

**Pull request merged**